src/Components/Blazor/WebAssembly.Authentication/src/Services/SignOutSessionStateManager.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// Handles CSRF protection for the logout endpoint.
+    /// </summary>
+    public class SignOutSessionStateManager
```

Comment:
I notice that the usage of this really complicates the `Authentication.razor` page in your testassets project. It somewhat undermines the idea that `AuthenticationManager` encapsulates everything needed.

Could you clarify why `AuthenticationManager` isn't able to deal with this as an internal implementation detail? Why do we need to expose public API relating to validating the sign out flow?

Reply:
I've integrated this into the `RemoteAuthenticatorViewCore` the reason I wanted outside is that I didn't want to couple the RAVC with the idea that you need to have a link for logging out exposed, but there's a way to opt out of this validation by extending the class and overriding ValidateState.